### PR TITLE
add --ampls-resource-id and --enable-high-log-scale-mode options

### PR DESCRIPTION
**Pull request by ganga1980**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -2333,7 +2356,9 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
                    dns_zone_resource_ids=None,
                    no_wait=False,  # pylint: disable=unused-argument
                    enable_syslog=False,
-                   data_collection_settings=None):
+                   data_collection_settings=None,
+                   ampls_resource_id=None,
+                   enable_high_log_scale_mode=False):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2361]: It seems to me that these parameters are not really used (passed to ensure_container_insights_for_monitoring and create some resources) but are just discarded here. And it looks like the same thing happens in azure-cli.

Note there are two commands that can be used to enable an addon in aks-preview, namely `az aks enable-addons` and `az aks addon enable`. They have completely different code paths. In azure-cli, only `az aks enable-addons` is supported.

**Review by FumingZhang — COMMENTED** [review]
please add some live tests for the newly added options.

File: src/aks-preview/azext_aks_preview/custom.py (hunk 1 above)
Comment (ganga1980) [reply src/aks-preview/azext_aks_preview/custom.py]: > It seems to me that these parameters are not really used (passed to ensure_container_insights_for_monitoring and create some resources) but are just discarded here. And it looks like the same thing happens in azure-cli.
> 
> Note there are two commands that can be used to enable an addon in aks-preview, namely `az aks enable-addons` and `az aks addon enable`. They have completely different code paths. In azure-cli, only `az aks enable-addons` is supported.

removed unused parameters here and as well azure-cli.

**Comment by yanzhudd** [comment]
please fix the CI issues

**Comment by ganga1980** [comment]
Hi, @yanzhudd , @FumingZhang   - CI test failures not to related to this PR and CI test failures related to this test "FAILED src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py::AzureKubernetesServiceScenarioTest::test_aks_create_with_pod_ip_allocation_mode_static_block" and I see same CI tests failures in other PRs also.

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=97176265&view=results) to validate the change.

- test_aks_create_with_monitoring_aad_auth_with_highlogscale
- test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls
- test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls_with_highlogscale

failed with error
> - azure.cli.testsdk.exceptions.JMESPathCheckAssertionError: Query 'properties.dataCollectionEndpointId' doesn't yield expected value '/subscriptions/79a7390d-3a85-432d-9f6f-a11a703c8b83/resourceGroups/clitestfk7a24nt2b/providers/Microsoft.Insights/dataCollectionEndpoints/MSCI-ingest-westus2-cliakstestgm5hrh', instead the actual value is '/subscriptions/79a7390d-3a85-432d-9f6f-a11a703c8b83/resourceGroups/clitestfk7a24nt2b/providers/Microsoft.Insights/dataCollectionEndpoints/MSCI-westus2-cliakstestgm5hrh'. Data: ...
> - azure.cli.core.azclierror.HTTPError: Not Found({"error":{"code":"ResourceNotFound"}})

**Comment by FumingZhang** [comment]
> lgtm
> 
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=97176265&view=results) to validate the change.
> 
> * test_aks_create_with_monitoring_aad_auth_with_highlogscale
> * test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls
> * test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls_with_highlogscale
> 
> failed with error
> 
> > * azure.cli.testsdk.exceptions.JMESPathCheckAssertionError: Query 'properties.dataCollectionEndpointId' doesn't yield expected value '/subscriptions/79a7390d-3a85-432d-9f6f-a11a703c8b83/resourceGroups/clitestfk7a24nt2b/providers/Microsoft.Insights/dataCollectionEndpoints/MSCI-ingest-westus2-cliakstestgm5hrh', instead the actual value is '/subscriptions/79a7390d-3a85-432d-9f6f-a11a703c8b83/resourceGroups/clitestfk7a24nt2b/providers/Microsoft.Insights/dataCollectionEndpoints/MSCI-westus2-cliakstestgm5hrh'. Data: ...
> > * azure.cli.core.azclierror.HTTPError: Not Found({"error":{"code":"ResourceNotFound"}})

Requeued [test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=97179558&view=results)

**Comment by FumingZhang** [comment]
Requeued another [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98236519&view=results)

**Comment by FumingZhang** [comment]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98698613&view=results)
- test_aks_create_with_monitoring_aad_auth_with_highlogscale
- test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls
- test_aks_create_with_private_cluster_with_monitoring_aad_auth_msi_with_ampls_with_highlogscale

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -17,6 +17,11 @@ Pending
 ++++++++
 * [BREAKING CHANGE]: Remove support for `az aks get-os-options` command.
 
+6.0.0b2
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:20]: Please choose a new version number like 7.0.0b2

Comment (FumingZhang) [reply src/aks-preview/HISTORY.rst:20]: And includes the content listed in section Pending

Comment (ganga1980) [reply src/aks-preview/HISTORY.rst]: done.

**Comment by FumingZhang** [comment]
[Live test]((https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=99330531&view=results)) failed

**Comment by FumingZhang** [comment]
Please fix style issues.
https://dev.azure.com/azclitools/public/_build/results?buildId=177380&view=logs&j=506b8f13-8269-5210-246e-a3964e53141e&t=db7247cb-300d-5b59-fd6c-6258fad20b89&l=152

> ERROR: ************* Module azext_aks_preview._helpers
> src/aks-preview/azext_aks_preview/_helpers.py:346:11: W0718: Catching too general exception Exception (broad-exception-caught)
> src/aks-preview/azext_aks_preview/_helpers.py:338:19: W1508: os.getenv default type is builtins.bool. Expected str or None. (invalid-envvar-default)

Requeued [live test](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=99442109&view=results)

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -3733,10 +3734,43 @@ def construct_mc_profile_preview(self, bypass_restore_defaults: bool = False) ->
...
+            logger.debug("failed to get the value of monitor_options: %s", ex)
+            monitor_options_specified = any(
+                (
+                    self.__raw_parameters.get("ampls_resource_id"),
+                    self.__raw_parameters.get("enable_high_log_scale_mode"),
+                )
+            )
+        finally:
+            if monitor_options_specified and (
+                core_version < "2.63.0" and
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3765]: not blocking: it would be better to do the comparison semantically

**Review by FumingZhang — COMMENTED** [review]
lgtm

Please update VERSION (in file `src/aks-preview/setup.py) `to `7.0.0b3`

**Review by FumingZhang — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**